test/e2e/common/setup-kubestellar.sh

```diff
@@ -13,188 +13,210 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -x # echo so users can understand what is happening
-set -e # exit on error
 
-use_release=false
+###############################################################################
+# Combined setup + end-to-end test for KubeStellar with Argo CD integration
+###############################################################################
+
+set -o errexit   # exit on any command failure
+set -o nounset   # exit on use of un-set variable
+set -o pipefail  # pipeline fails if any sub-command fails
+set -o errtrace  # trap ERR in functions and subshells
+shopt -s inherit_errexit  # propagate set -e into subshells
+set -x            # echo commands so users can follow along
+
+
+# ─────────────────────────────────────────────────────────────────────────────
+# Defaults
+# ─────────────────────────────────────────────────────────────────────────────
+ENV=kind
+ARGOCD_DOMAIN=argocd.localtest.me
+TEST_TIMEOUT=300          # seconds
+USE_RELEASE=false
 KUBESTELLAR_CONTROLLER_MANAGER_VERBOSITY=5
 TRANSPORT_CONTROLLER_VERBOSITY=5
 CLUSTER_SOURCE=kind
 HOSTING_CONTEXT=kind-kubeflex
 
-while [ $# != 0 ]; do
+TEST_APP_REPO="https://github.com/argoproj/argocd-example-apps.git"
+TEST_APP_PATH="guestbook"
+
+
+# ─────────────────────────────────────────────────────────────────────────────
+# Colourised logging helpers
+# ─────────────────────────────────────────────────────────────────────────────
+RED='\033[0;31m'; GREEN='\033[0;32m'
+YELLOW='\033[1;33m'; BLUE='\033[0;34m'
+NC='\033[0m' # no colour
```

Comment:
If this line said `NC=$'\033[0m' # no colour` then the later `echo` commands would not need the dangerous `-e`.